### DISCUSSION TITLE
trying to access enums from c# .dll with python > 3.9 and pythonnet

### DISCUSSION THREAD

**Discussion by oasisjpeg** [pythonnet/pythonnet#2185] [Q&A]
### Environment

-   Pythonnet version: 3.0.1
-   Python version: 3.11.3
-   Operating System: Windows
-   .NET Runtime: 4.8

### Details

-   Describe what you were trying to get done.

i am a really newbie when it comes to python and c#

i have got a project that is using python 3.8.5 and pythonnet 2.5.0. so now i am trying to get it working on the newest python 3.11.3 and pythonnet 3.0.1.

on the old version it was easy to access the enums inside the .dll and it works everything as it should. but now after updating to the new version i get a weird error message that i don't really understand.

first of all this is my error:
`No method matches given arguments for Testing.function: (<enum 'myEnum'>)`

-   What commands did you run to trigger this issue? If you can provide a
following i am going to add my rather simple replica that will force the same error like on my bigger problem:

in one file i try to call my other python script with following code:

```python
import otherPythonfile
from otherPythonfile import myEnum as myEnum


if __name__ == "__main__":

    otherPythonfile.function(myEnum.range_300_V)
```
the otherPythonfile.py has following code:
```python
import os as __os
import clr as __clr
import System as __System


PATH_TEST = __os.path.dirname(__os.path.abspath(__file__))
try:
    from .enums.myEnum import myEnum

except ImportError:  # normal Python environment
    from enums.myEnum import myEnum

__clr.AddReference(PATH_TEST + "//resources//cs_backend")
import cs_backend as __csb  # noqa
__test = __csb.Testing()


def function(
    myEnum: myEnum,
):  
    __test.function(myEnum)
```
when calling my "__test.function", that is just a Console.WriteLine inside my C#, i get the error on the image above.

in the changelog there stands following log: "BREAKING: disabled implicit conversion from C# enums to Python int and back. One must now either use enum members (e.g. MyEnum.Option), or use enum constructor (e.g. MyEnum(42) or MyEnum(42, True) when MyEnum does not have a member with value 42)." Does this have to do something with my problem?

**Comment by DeniseSkidmore** [comment]
Where is myEnum defined?

**Comment by lostmsu** [comment]
What's the signature of `function` in C#?
